### PR TITLE
Features/amend product action boxes

### PR DESCRIPTION
**Pull request by ssrajan**
1. Renamed the Production Actions to 'ENC Actions'
2. New controls for SNC actions are created.
3. All the fields for Product Actions are kept as it is, but the reference to it in the error messages are changed to ENC Actions
4. Implemented SNC actions without lookup table for imported products.
5. New Test cases are added for SNC actions

### CODE REVIEW

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml [hunk 1]
```diff
@@ -74,6 +75,8 @@
     <div id="recordProductAction"></div>
     <div id="recordProductActionError"></div>
 
+    <div id="recordSncAction"></div>
+
```
Comment (SamirHassoun) [inline src/Portal/Portal/Pages/DbAssessment/Assess.cshtml:79]: You'll need the `<div id="recordSncActionError"></div>` and populate accordingly

File: src/Portal/Portal/wwwroot/js/RecordSncAction.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                setCreateHandler();
+                setSncActionedCheckboxHandler();
+                update();
+
+                setControlState();
+
+                setImpactedProductHandler();
+            },
+            error: function (error) {
+                $("#recordSncActionError")
```
Comment (SamirHassoun) [inline src/Portal/Portal/wwwroot/js/RecordSncAction.js:36]: You'll need to add `<div id="recordSncActionError"></div> ` to Assess\Verify pages and populate accordingly

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 1 above)
Comment (ssrajan) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml:80]: Added the new section

File: src/Portal/Portal/wwwroot/js/RecordSncAction.js (hunk 2 above)
Comment (ssrajan) [reply src/Portal/Portal/wwwroot/js/RecordSncAction.js:36]: changes done in access and verify pages

File: src/Portal/Portal/Pages/DbAssessment/_RecordSncAction.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        private async Task SetSncActionFromDb()
+        {
+            SncActions = await _dbContext.SncAction
+                .Include(x => x.SncActionType)
+                .Where(p => p.ProcessId == ProcessId)
+                .ToListAsync();
+
+            if (SncActions == null || SncActions.Count == 0)
```
Comment (benbhall) [inline src/Portal/Portal/Pages/DbAssessment/_RecordSncAction.cshtml.cs:64]: The LINQ query returns a new `List<SncAction>` so probably just need the following, if you need a single blank `SncAction` in there:

```csharp
if (!SncActions.Any()) SncActions.Add(new SncAction());
```

**Review by benbhall — APPROVED** [review]

**Review by SamirHassoun — APPROVED** [review]

**Pull request merged**